OSDOCS-20033: Kueue 1.4 and DRA (#113996)
openshift/openshift-docs: modules/kueue-dra-quota-management.adoc

{kueue-name} provides two approaches for managing DRA device quota:

`ResourceClaimTemplate`:: The default approach. Workloads explicitly reference a `ResourceClaimTemplate` that defines device requirements. Administrators configure `deviceClassMappings` in the Kueue CR to map each `DeviceClass` to a logical resource name for quota tracking. Use this approach when workloads need fine-grained control over device selection, such as targeting a specific GPU model or architecture using CEL selectors.

Extended resources:: A simplified alternative that allows workloads to use standard Kubernetes `resources.requests` syntax, for example, `nvidia.com/gpu: "1"`, instead of explicitly creating DRA objects. When a `DeviceClass` includes the `spec.extendedResourceName` field, the Kubernetes scheduler automatically generates `ResourceClaim` objects. {kueue-name} detects this and charges quota only once, preventing double counting. Use this approach when you want the simplest possible user experience and backward compatibility with existing workload YAML.

> 

Reviewer: The "preventing double counting" part might be worth simplifying — it could give the impression that there's a risk of the same GPU being charged twice, when the behavior is actually about how kueue resolves  the quota key. Suggest removing that sentence and adding a separate note:

[NOTE]
If a `DeviceClass` with `extendedResourceName` also appears in `deviceClassMappings`, {kueue-name} uses the mapped logical name from `deviceClassMappings` for quota instead of the extended resource name,  unifying quota accounting across both paths.

Author: Sentence removed, added the Note.

// Module included in the following assemblies:
//
// * ai_workloads/kueue/kueue-dra-integrating-dynamic-resource-allocation.adoc

:_mod-docs-content-type: CONCEPT
[id="kueue-dra-quota-management_{context}"]
= DRA quota management overview

[role="_abstract"]

> Dynamic Resource Allocation (DRA) is a Kubernetes framework that provides structured discovery and allocation of specialized hardware such as GPUs. DRA drivers publish device information through `ResourceSlice` objects, and administrators group devices into named categories using `DeviceClass` objects.

Reviewer 2: This short description is just over 300 characters. To meet DITA requirements, it needs to be 300 characters or fewer.

Without {kueue-name} DRA integration, GPU requests made through DRA are invisible to quota management. {kueue-name} cannot account for these requests when admitting workloads, which can result in teams exceeding their GPU allocation.

> {kueue-name} provides two approaches for managing DRA device quota:

Reviewer 2: Check the preview here. It looks a little odd to me that the formatting of the two approaches doesn't match, but maybe that's okay?

> `ResourceClaimTemplate`:: The default approach. Workloads explicitly reference a `ResourceClaimTemplate` that defines device requirements. Administrators configure `deviceClassMappings` in the Kueue CR to map each `DeviceClass` to a logical resource name for quota tracking. Use this approach when workloads need fine-grained control over device selection, such as targeting a specific GPU model or architecture using CEL selectors.

Reviewer 2: ```suggestion
`ResourceClaimTemplate`:: The default approach. Workloads explicitly reference a `ResourceClaimTemplate` object that defines device requirements. Administrators configure `deviceClassMappings` in the Kueue CR to map each `DeviceClass` object to a logical resource name for quota tracking. Use this approach when workloads need fine-grained control over device selection, such as targeting a specific GPU model or architecture using CEL selectors.
```
As much as possible, try to add whatever the appropriate noun is after an object reference.

> Extended resources:: A simplified alternative that allows workloads to use standard Kubernetes `resources.requests` syntax, for example, `nvidia.com/gpu: "1"`, instead of explicitly creating DRA objects. When a `DeviceClass` includes the `spec.extendedResourceName` field, the Kubernetes scheduler automatically generates `ResourceClaim` objects. Use this approach when you want the simplest possible user experience and backward compatibility with existing workload YAML.

Reviewer 2: Similar comment here about providing the appropriate noun after an object reference.

```suggestion
Extended resources:: A simplified alternative that allows workloads to use standard Kubernetes `resources.requests` syntax, for example, `nvidia.com/gpu: "1"`, instead of explicitly creating DRA objects. When a `DeviceClass` object includes the `spec.extendedResourceName` field, the Kubernetes scheduler automatically generates `ResourceClaim` objects. Use this approach when you want the simplest possible user experience and backward compatibility with existing workload YAML.
```

+
[NOTE]
====

> If a `DeviceClass` with `extendedResourceName` also appears in `deviceClassMappings`, {kueue-name} uses the mapped logical name from `deviceClassMappings` for quota instead of the extended resource name, unifying quota accounting across both paths.

Reviewer 2: ```suggestion
If a `DeviceClass` object with the `extendedResourceName` parameter also appears in the `deviceClassMappings` specification, {kueue-name} uses the mapped logical name from the `deviceClassMappings` specification for quota instead of the extended resource name, unifying quota accounting across both paths.
```
I made a guess at the correct object references here; feel free to change them to whatever is accurate.

====

> For clusters with partitionable devices such as NVIDIA Multi-Instance GPU (MIG), {kueue-name} can also charge quota in capacity units, such as GPU memory, rather than device count. Partitionable devices use `ResourceClaimTemplates` with CEL selectors to target specific partition profiles, and require administrators to configure counter-based `sources` in `deviceClassMappings`. This capability requires {product-title} 4.22 or later.

Reviewer 2: ```suggestion
For clusters with partitionable devices, such as NVIDIA Multi-Instance GPU (MIG), {kueue-name} can also charge quota in capacity units, such as GPU memory, rather than device count. Partitionable devices use `ResourceClaimTemplate` objects with CEL selectors to target specific partition profiles, and require administrators to configure counter-based `sources` in `deviceClassMappings`. This capability requires {product-title} 4.22 or later.
```
Same goes here for using the appropriate object references.
